libpod/container_log_linux.go

```diff
@@ -34,18 +35,24 @@ func (c *Container) readFromJournal(ctx context.Context, options *logs.LogOption
 	var config journal.JournalReaderConfig
 	if options.Tail < 0 {
 		config.NumFromTail = 0
+	} else if options.Tail == 0 {
+		config.NumFromTail = math.MaxUint64
 	} else {
 		config.NumFromTail = uint64(options.Tail)
 	}
-	config.Formatter = journalFormatter
+	if options.Multi {
+		config.Formatter = journalFormatterWithID
+	} else {
+		config.Formatter = journalFormatter
+	}
 	defaultTime := time.Time{}
 	if options.Since != defaultTime {
 		// coreos/go-systemd/sdjournal doesn't correctly handle requests for data in the future
 		// return nothing instead of falsely printing
 		if time.Now().Before(options.Since) {
 			return nil
 		}
-		config.Since = time.Since(options.Since)
+		config.Since = -time.Since(options.Since)
```

Comment:
OK, I'll fess up to my ignorance.  What does the minus `-` sign do here?

Reply:
It just makes the value negative. The journald package we're using expects a negative `time.Duration` for times in the past and positive `time.Duration` for times in the future. Without the negative, a `--since 10m` ends up meaning "since 10 minutes in the future", which was why the `log --since` tests with journald weren't working

Reply:
I can edit the commit message or add a comment to say this

Reply:
Thanks for the 411 @ashley-cui.  I'll merge now, but if you get a chance later, a small code comment there would be helpful.